unitTests/curvatureResistance/curvatureResistance.py

```diff
@@ -0,0 +1,285 @@
+import json
+import matplotlib.pyplot as plt
+import numpy as np
+import os
+import sys
+import tempfile
+import unittest
+
+sys.path.append('../..')
+
+from efficiency import totalLossesFunction
+from ocp import casadiSolver
+from train import Train
+from track import Track
+from utils import postProcessDataFrame
+
+
+class TestCurvatureResistance(unittest.TestCase):
+
+    TRACK = "../../tracks/00_stationX_stationY.json"
+
+    CONSTANT_K = 1/300 # [m^-1]
+
+    DEFAULT_TRIP_FINAL_POSITION = 3475 # [m]
+
+    DEFAULT_SPEED_AT_START = 1 # [m/s]
+
+    DEFAULT_SPEED_AT_END = 1 # [m/s]
+    
+    THRESHOLD_VELOCITY_ERROR_MINIMUM_TIME_PROBLEM = 1e-3 # adimensional quantity
+
+    THRESHOLD_ENERGY_ERROR_MINIMUM_ENERGY_PROBLEM = 5e-2 # adimensional quantity
+
+    @classmethod
+    def setUpClass(cls):
+        '''
+        Based on the dataset contained in TRACK, two temporary track datasets are created.
+        In each test we solve optimal control problems for both tracks and we compare their results.
+        '''
+
+        cls.fileNoCurvature = tempfile.NamedTemporaryFile(delete=False, suffix='.json', mode='w+')
+
+        cls.fileConstantCurvature = tempfile.NamedTemporaryFile(delete=False, suffix='.json', mode='w+')
+
+        try:
+
+            with open(cls.TRACK) as file:
+
+                originalTrack = json.load(file)
+
+            trackNoCurvature = originalTrack
+
+            del trackNoCurvature['curvatures']
+
+            json.dump(trackNoCurvature, cls.fileNoCurvature, indent=4)
+
+            cls.fileNoCurvature.flush()
+
+            trackConstantCurvature = trackNoCurvature
+
+            trackConstantCurvature["curvatures"] = {"units": {"position": "m", "radius at start": "m", "radius at end": "m" }, 
+                                                    "values": [[0.0, str(1/cls.CONSTANT_K), str(1/cls.CONSTANT_K)]]}
+            
+            json.dump(trackConstantCurvature, cls.fileConstantCurvature, indent=4)
+
+            cls.fileConstantCurvature.flush()
+
+        except Exception:
+
+            cls.tearDownClass()
+
+            raise unittest.SkipTest("Something went wrong when attempting to create test tracks. ABORTING ALL TESTS!")
+        
+        
+    @classmethod
+    def tearDownClass(cls):
+
+        cls.fileNoCurvature.close(), cls.fileConstantCurvature.close()
+
+        os.unlink(cls.fileNoCurvature.name), os.unlink(cls.fileConstantCurvature.name)
+
+    
+
+    def specificCurvatureResistanceForce(cls, g, rho):
```

Comment:
since all functions that follow are not class methods, shouldn't we use 'self' instead of 'cls' as first input?